x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/use_alert_suppression.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback } from 'react';
+import { isSuppressibleAlertRule } from '../../../../../common/detection_engine/utils';
+import { SuppressibleAlertRules } from '../../../../../common/detection_engine/constants';
+import type { ExperimentalFeatures } from '../../../../../common';
+import { useIsExperimentalFeatureEnabled } from '../../../../common/hooks/use_experimental_features';
+import type { RuleResponse } from '../../../../../common/api/detection_engine/model/rule_schema';
+
+export interface UseAlertSuppressionReturn {
+  isSuppressionEnabled: boolean;
+}
+
+export const useAlertSuppression = (rule: Partial<RuleResponse>): UseAlertSuppressionReturn => {
```

Comment:
Good work on this hook, which can abstract alert suppression + feature flag logic.

Can it be also used on rule/edit create page?

Reply:
component that used on rule create/edit: `x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/index.tsx`

Reply:
[055475d](https://github.com/elastic/kibana/pull/175089/commits/055475d651850c32bc110c384914e918689b07a2)